nnvm/src/top/tensor/reduce.cc

```diff
@@ -116,6 +133,17 @@ inline void AxesParamParser(nnvm::NodeAttrs* attrs) {
   .set_num_inputs(1)                                                    \
   .set_num_outputs(1)
 
+#define NNVM_REGISTER_COLLAPSE_OP(op)                                     \
+  NNVM_REGISTER_OP(collapse_ ## op)                                      \
+  .add_argument("data", "Tensor", "The input")                          \
+  .add_argument("as", "Tensor", "The reference")                          \
+  .add_arguments(ReduceParam::__FIELDS__())                             \
+  .set_attr_parser(AxesParamParser<ReduceParam>)                        \
+  .set_attr<FInferShape>("FInferShape", CollapseShape)                    \
+  .set_attr<FInferType>("FInferType", ElemwiseType<2, 1>)               \
```

Comment:
We need FCorrectLayout attribute to for correct layout pass. An example for elemwise binary op https://github.com/dmlc/tvm/blob/master/nnvm/src/top/elemwise_op_common.h#L301.